presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMaterializedViewLogicalPlanner.java

```diff
@@ -1471,6 +1474,294 @@ public void testMaterializedViewForJoin()
+        }
+        finally {
+            queryRunner.execute("DROP MATERIALIZED VIEW IF EXISTS " + lineItemView1);
+            queryRunner.execute("DROP MATERIALIZED VIEW IF EXISTS " + lineItemView2);
+            queryRunner.execute("DROP MATERIALIZED VIEW IF EXISTS " + suppliersView);
+            queryRunner.execute("DROP TABLE IF EXISTS " + lineItemTable);
+            queryRunner.execute("DROP TABLE IF EXISTS " + supplierTable);
+        }
+    }
+
+    @Test
+    public void testSubqueryMaterializedViewAggregateWithAndJoin()
+    {
+        Session queryOptimizationWithMaterializedView = Session.builder(getSession())
+                .setSystemProperty(QUERY_OPTIMIZATION_WITH_MATERIALIZED_VIEW_ENABLED, "true")
+                .setSystemProperty(PREFER_PARTIAL_AGGREGATION, "false")
+                .build();
+        QueryRunner queryRunner = getQueryRunner();
+
+        String supplierTable = "supplier_partitioned";
+        String lineItemTable = "lineitem_partitioned";
+        String lineItemView = "test_lineitem_view";
+
+        try {
+            queryRunner.execute(format("CREATE TABLE %s " +
+                            "WITH (partitioned_by = array['ds']) " +
+                            "AS SELECT *, '2021-07-11' AS ds " +
+                            "FROM supplier " +
+                            "WHERE suppkey < 10000 " +
+                            "UNION ALL " +
+                            "SELECT *, '2021-07-12' AS ds " +
+                            "FROM supplier " +
+                            "WHERE suppkey < 10000",
+                    supplierTable));
+
+            queryRunner.execute(format("CREATE TABLE %s " +
+                            "WITH (partitioned_by = array['ds']) " +
+                            "AS SELECT *, '2021-07-11' AS ds " +
+                            "FROM lineitem " +
+                            "WHERE quantity > 1 " +
+                            "UNION ALL " +
+                            "SELECT *, '2021-07-12' AS ds " +
+                            "FROM lineitem " +
+                            "WHERE quantity > 1",
+                    lineItemTable));
+
+            assertUpdate(format("CREATE MATERIALIZED VIEW %s " +
+                            "WITH (partitioned_by = ARRAY['ds']) " +
+                            "AS SELECT MIN(extendedprice) AS min_price, partkey, ds " +
+                            "FROM %s " +
+                            "GROUP BY partkey, ds ",
+                    lineItemView, lineItemTable));
+
+            assertTrue(getQueryRunner().tableExists(getSession(), lineItemView));
+
+            setReferencedMaterializedViews((DistributedQueryRunner) queryRunner, lineItemTable, ImmutableList.of(lineItemView));
+
+            assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE ds='2021-07-11'", lineItemView), 2000);
+            assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE ds='2021-07-12'", lineItemView), 2000);
```

Comment:
I think at this point we are always refreshing everything for most cases. By default in the tests, we should not refresh everything as it would be the case in the reality. Can you fix tests to make sure at most places Materialized view is partially refreshed?